src/AdoNetCore.AseClient/Internal/TypeMap.cs

```diff
@@ -340,5 +342,96 @@ public static AseDbType GetAseDbType(FormatItem format)
+            {DbType.DateTime, AseDbType.DateTime},
+            {DbType.DateTime2, AseDbType.DateTime},
+            {DbType.DateTimeOffset, AseDbType.Unsupported},
+            {DbType.Decimal, AseDbType.Decimal},
+            {DbType.Double, AseDbType.Double},
+            {DbType.Guid, AseDbType.Unsupported},
+            {DbType.Int16, AseDbType.SmallInt},
+            {DbType.Int32, AseDbType.Integer},
+            {DbType.Int64, AseDbType.BigInt},
+            {DbType.Object, AseDbType.Binary},
+            {DbType.SByte, AseDbType.SmallInt}, //Was Unsupported
+            {DbType.Single, AseDbType.Real},
+            {DbType.String, AseDbType.UniVarChar},
+            {DbType.StringFixedLength, AseDbType.UniChar},
+            {DbType.Time, AseDbType.Time},
+            {DbType.UInt16, AseDbType.UnsignedSmallInt},
+            {DbType.UInt32, AseDbType.UnsignedInt},
+            {DbType.UInt64, AseDbType.UnsignedBigInt},
+            {DbType.VarNumeric, AseDbType.Numeric},
+            {DbType.Xml, AseDbType.Unsupported},
+        };
+
+        public static AseDbType GetAseDbType(DbType dbType)
+        {
+            return DbTypeToAseDbTypeMap.ContainsKey(dbType)
+                ? DbTypeToAseDbTypeMap[dbType]
+                : default(AseDbType);
+        }
+
+        private static readonly Dictionary<AseDbType, DbType> AseDbTypeToDbTypeMap = new Dictionary<AseDbType, DbType>
+        {
+            //{AseDbType.BigDateTime, DbType.DateTime}, //same value as DateTime
+            {AseDbType.BigInt, DbType.Int64},
+            {AseDbType.Binary, DbType.Binary},
+            {AseDbType.Bit, DbType.Boolean},
+            {AseDbType.Char, DbType.AnsiStringFixedLength},
+            {AseDbType.Date, DbType.Date},
+            {AseDbType.DateTime, DbType.DateTime},
+            {AseDbType.Decimal, DbType.Decimal},
+            {AseDbType.Double, DbType.Double},
+            {AseDbType.Image, DbType.Binary},
+            {AseDbType.Integer, DbType.Int32},
+            {AseDbType.Money, DbType.Currency},
+            {AseDbType.NChar, DbType.AnsiStringFixedLength},
+            {AseDbType.Numeric, DbType.VarNumeric},
+            {AseDbType.NVarChar, DbType.AnsiString},
+            {AseDbType.Real, DbType.Single},
+            {AseDbType.SmallDateTime, DbType.DateTime},
+            {AseDbType.SmallInt, DbType.Int16},
+            {AseDbType.SmallMoney, DbType.Currency},
+            {AseDbType.Text, DbType.AnsiString},
+            {AseDbType.Time, DbType.Time},
+            {AseDbType.TimeStamp, DbType.Binary},
+            {AseDbType.TinyInt, DbType.Byte},
+            {AseDbType.UniChar, DbType.StringFixedLength},
+            {AseDbType.Unitext, DbType.String},
+            {AseDbType.UniVarChar, DbType.String},
+            {AseDbType.UnsignedBigInt, DbType.UInt64},
+            {AseDbType.UnsignedInt, DbType.UInt32},
+            {AseDbType.UnsignedSmallInt, DbType.UInt16},
+            {AseDbType.Unsupported, DbType.Guid},
```

Comment:
Why are we mapping Unsupported to Guid?

Reply:
For consistency with the reference driver -- It appears to treat `Guid` as a synonym for unsupported.

If the user explicitly sets the DbType to Guid, then they can still use the Guid type (as demonstrated by `SimpleQueryTests.SelectGuid_Parameter_ShouldWork`)